feature_layers/dynamic-entity-layer/src/main/java/com/esri/samples/dynamic_entity_layer/DynamicEntityLayerSample.java

```diff
@@ -0,0 +1,307 @@
+      var streamServiceUrl = "https://realtimegis2016.esri.com:6443/arcgis/rest/services/SandyVehicles/StreamServer";
+      var streamService = new ArcGISStreamService(streamServiceUrl);
+
+      // add a filter for the data source to limit the amount of data received by the application
+      var streamServiceFilter = new ArcGISStreamServiceFilter();
+      streamServiceFilter.setGeometry(utahSandyEnvelope);
+      streamServiceFilter.setWhereClause("speed > 0");
+      streamService.setFilter(streamServiceFilter);
+
+      // set a duration of five minutes for how long observation data is stored in the data source
+      streamService.getPurgeOptions().setMaximumDuration(5);
+
+      // create a layer to display the data from the stream service
+      dynamicEntityLayer = new DynamicEntityLayer(streamService);
+
+      // create unique value renderers for customizing the dynamic entity layer appearance
+      var uniqueValueRenderers = createUniqueRenderers();
+
+      // set the dynamic entity and previous observation renderers
+      dynamicEntityLayer.setRenderer(uniqueValueRenderers.get(0));
+      dynamicEntityLayer.getTrackDisplayProperties().setPreviousObservationRenderer(uniqueValueRenderers.get(1));
+
+      // create a simple line renderer and set the track line renderer to it
+      var trackLineRenderer = new SimpleRenderer(new SimpleLineSymbol(SimpleLineSymbol.Style.SOLID, Color.LIGHTGRAY, 2));
+      dynamicEntityLayer.getTrackDisplayProperties().setTrackLineRenderer(trackLineRenderer);
+
+      // update the label with the service connection status
+      connectionStatusLabel.textProperty().bind(Bindings.createStringBinding(() ->
+        "Status: " + streamService.connectionStatusProperty().asString().get(), streamService.connectionStatusProperty()));
+
+      // update the button text with the service connection status
+      connectionButton.textProperty().bind(Bindings.createStringBinding(() ->
+        streamService.getConnectionStatus() == ConnectionStatus.CONNECTED ?
+          "Disconnect" : "Connect", streamService.connectionStatusProperty()));
+
+      // toggle the service connection on button press
+      connectionButton.setOnAction(a -> {
+        if (streamService.getConnectionStatus() == ConnectionStatus.CONNECTED) {
+          streamService.disconnectAsync();
+        } else {
+          streamService.connectAsync();
+        }
+      });
+
+      // enable or disable track line visibility with the checkbox
+      trackLinesChkBox.selectedProperty().bindBidirectional(dynamicEntityLayer.trackDisplayPropertiesProperty().get().
+        showTrackLineProperty());
+
+      // enable or disable visibility of previous observation tracks with the checkbox
+      observationsChkBox.selectedProperty().bindBidirectional(dynamicEntityLayer.trackDisplayPropertiesProperty().get().
+        showPreviousObservationsProperty());
+
+      // update label with the maximum number of observations set in the slider
```

Comment:
Something that could be useful for each of the blocks of UI here, is to have a general comment first like
`// configure a Slider to control the number of observations displayed per track`
You could do the same for each section of functionality